packages/renderer/src/lib/kube/KubeApplyYAML.spec.ts

```diff
@@ -0,0 +1,216 @@
+/*********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************/
+
+import '@testing-library/jest-dom/vitest';
+
+import type { Context } from '@kubernetes/client-node';
+import { fireEvent, render, waitFor, within } from '@testing-library/svelte';
+import userEvent from '@testing-library/user-event';
+import { beforeEach, describe, expect, test, vi } from 'vitest';
+
+import KubeApplyYAML from './KubeApplyYAML.svelte';
+
+// Mock the router
+vi.mock('tinro', () => ({
+  router: {
+    goto: vi.fn(),
+  },
+}));
+
+vi.mock('../editor/MonacoEditor.svelte', async () => ({
+  default: (await import('../editor/MonacoEditorTestDouble.svelte')).default,
+}));
+
+const contexts: Context[] = [
+  {
+    cluster: 'cluster1',
+    user: 'user1',
+    name: 'context1',
+    namespace: 'ns1',
+  },
+  {
+    cluster: 'cluster2',
+    user: 'user2',
+    name: 'context2',
+    namespace: 'ns2',
+  },
+];
+
+beforeEach(() => {
+  vi.clearAllMocks();
```

Comment:
```suggestion
  vi.resetAllMocks();
```

`resetAllMocks()` is prefered over `clearAllMocks()`